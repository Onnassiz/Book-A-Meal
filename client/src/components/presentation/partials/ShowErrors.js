import React from 'react';
import empty from 'is-empty';
import PropTypes from 'prop-types';


const ShowErrors = props => (
  <div className="show-errors">
    <ul>
      {Object.keys(props.clientErrors).map(item =>
        <li key={item}>{props.clientErrors[item]}</li>)}
      {empty(props.serverErrors) ? '' : Object.keys(props.serverErrors).map(item =>
        <li key={item}>{props.serverErrors[item]}</li>)}
    </ul>
  </div>
);

ShowErrors.propTypes = {
  serverErrors: PropTypes.array.isRequired,
  clientErrors: PropTypes.array.isRequired,
};

export default ShowErrors;
