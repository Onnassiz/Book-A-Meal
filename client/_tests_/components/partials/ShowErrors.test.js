import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ShowErrors from '../../../src/components/presentation/partials/ShowErrors';

const props = {
  serverErrors: {
    message: 'Request failed',
  },
  clientErrors: {
    email: 'Email is invalid',
  },
};

const wrapper = shallow(<ShowErrors {...props} />);

describe('<ShowErrors />', () => {
  it('Should render error component successfully', () => {
    expect(wrapper.length).toBe(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
