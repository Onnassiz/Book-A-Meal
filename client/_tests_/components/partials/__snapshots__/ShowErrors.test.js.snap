// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShowErrors /> Should render error component successfully 1`] = `
<div
  className="show-errors"
>
  <ul>
    <li
      key="email"
    >
      Email is invalid
    </li>
    <li
      key="message"
    >
      Request failed
    </li>
  </ul>
</div>
`;
